### PR TITLE
Return symbols for all enum related methods

### PR DESCRIPTION
**Pull request by tjwallace**
Generated methods like [`#{column_name}_before_last_save`](https://github.com/1debit/chime-core/blob/5dade5e019009302e87855e0ed4144802500a762/app/models/user/active_record_hooks.rb#L289) were not returning symbols. This change forces those methods to return symbols, and overrides the generated value query methods (ie. `#status_pending?`) to compare symbols to symbols.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -7,10 +7,20 @@ on:
 
 jobs:
   rspec:
+    strategy:
```
Comment (jaredsmithse) [inline .github/workflows/tests.yml:10]: 👌

File: lib/sql_enum/enum_type.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module SqlEnum
+  # EnumType that always returns a symbol
+  class EnumType < ActiveRecord::Enum::EnumType
+    def cast(...)
```
Comment (jaredsmithse) [inline lib/sql_enum/enum_type.rb:4]: Is everything on 2.7 now?

Comment (tjwallace) [reply lib/sql_enum/enum_type.rb:4]: Ya that is a good question... Probably safer to just use normal args

Comment (tjwallace) [reply lib/sql_enum/enum_type.rb]: Switched to a normal arg

**Review by jaredsmithse — APPROVED** [review]

**Pull request merged**